### ISSUE TITLE
Add getProjection to Google Maps MapController

### ISSUE THREAD

**Issue by felangel** [flutter/flutter#37959]
## Use case

I want to be able to center the camera position in a google map when part of the map is being covered by a bottom sheet/drawer. In order to do this, I need to be able to convert `LatLng` to/from screen location.

For example, in Java you can convert the `LatLng` to/from a screen location via:

```java
map.getProjection().toScreenLocation(latLng);
map.getProjection().fromScreenLocation(point);
```

I don't think this is currently possible with the flutter google map implementation.

## Proposal

Add `getProjection` to the `MapController` in [google_maps_flutter](https://github.com/flutter/plugins/tree/master/packages/google_maps_flutter)

**Comment by Hixie** [comment]
cc @iskakaushik 

We likely won't have time to get to this in the coming month but this is on our radar now.

**Comment by kf6gpe** [comment]
@iskakaushik Does October still seem like a reasonable milestone for this given your work with `customer: fuchsia`?

**Comment by kf6gpe** [comment]
Discussed with @mjohnsullivan; this needs to land soon for `customer: vroom`.

**Comment by iskakaushik** [comment]
Hey @felangel , https://github.com/flutter/plugins/pull/2108 should expose translating between screen and latlng coordinates. I haven't exposed the full projection api just yet, the only method missing on the controller now is [pointForMeters](https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_projection.html#a7275229ff43ac7779c28979855c3c02f) and its android equivalent. Let me know if this is needed for your usecase and I can add it.

**Comment by felangel** [comment]
Thanks @iskakaushik! I don't think `pointForMeters` should be necessary for our use case. I'll give it a try over the next few days and let you know if we have any feedback. Thanks again for the quick response 👍

**Comment by iskakaushik** [comment]
Released `0.5.21+4` which should have this fix in.

**Comment by aswinkviswanath** [comment]
@iskakaushik @felangel is there any reason why pointForMeters was not added to the projection api's.  drawing dotted polylines in iOS makes use of this api since SDK doesn't support it out of the box. I can't find android equivalent for the same. any help on this would be great. thanks.
